connectivity/check/action.go

```diff
@@ -236,12 +241,26 @@ func (a *Action) ExecInPod(ctx context.Context, cmd []string) {
 	pod := a.src
 
 	a.Debug("Executing command", cmd)
-
-	output, err := pod.K8sClient.ExecInPod(ctx,
-		pod.Pod.Namespace, pod.Pod.Name, pod.Pod.Labels["name"], cmd)
-
 	cmdName := cmd[0]
 	cmdStr := strings.Join(cmd, " ")
+
+	var output bytes.Buffer
+	var err error
+	for i := 1; i < testCommandRetries; i++ {
+		output, err = pod.K8sClient.ExecInPod(ctx,
+			pod.Pod.Namespace, pod.Pod.Name, pod.Pod.Labels["name"], cmd)
+		// Check for inconclusive results.
+		if err == nil && strings.TrimSpace(output.String()) == "" {
+			a.Debugf("retrying command %s due to inconclusive results", cmdStr)
+			continue
+		}
+		break
+	}
+	// Check for inconclusive results.
+	if err == nil && strings.TrimSpace(output.String()) == "" {
+		a.Failf("inconclusive results: command %q was successful but without output", cmdStr)
+	}
+
 	a.cmdOutput = output.String()
```

Comment:
Each call to `output.String` will allocate (which might be quite a bit in cases the 
command output is large) and we're calling it two more times below. Let's store the allocated string in `a.cmdOutput` directly after executing the command and reuse it:

```suggestion
		output, err = pod.K8sClient.ExecInPod(ctx,
			pod.Pod.Namespace, pod.Pod.Name, pod.Pod.Labels["name"], cmd)
		a.cmdOutput = output.String()
		// Check for inconclusive results.
		if err == nil && strings.TrimSpace(a.cmdOutput) == "" {
			a.Debugf("retrying command %s due to inconclusive results", cmdStr)
			continue
		}
		break
	}
	// Check for inconclusive results.
	if err == nil && strings.TrimSpace(a.cmdOutput) == "" {
		a.Failf("inconclusive results: command %q was successful but without output", cmdStr)
	}

```